heron/api/src/java/com/twitter/heron/api/Config.java

```diff
@@ -452,6 +487,28 @@ public static void setTopologyExactlyOnceEnabled(Map<String, Object> conf, boole
     conf.put(Config.TOPOLOGY_EXACTLYONCE_ENABLED, String.valueOf(exactOnce));
   }
 
+  public static void setTopologyPackingAlgorithm(Map<String, Object> conf,
+                                                 PackingAlgorithmType type) {
```

Comment:
I think the intent (which makes sense to me) is that `api` is where  topology-level APIs are that topology developers might implement or specify (i.e., IBolt, ISpout, etc). The `spi` package is where system-level APIs live that would be implemented by administrating teams for a new scheduler or state manager for example.

I think the packing algorithms could be implemented or at least specified by topology authors at the topology level. Hence it makes sense for those interfaces to move to `api`, but not others perse.